src/io/iter_image_recordio_2.cc

```diff
@@ -388,44 +406,23 @@ inline void ImageRecordIOParser2<DType>::ParseChunk(dmlc::InputSplit::Blob * chu
           if (!std::is_same<DType, uint8_t>::value) {
             // normalize/mirror here to avoid memory copies
             // logic from iter_normalize.h, function SetOutImg
-
-            if (normalize_param_.mean_r > 0.0f || normalize_param_.mean_g > 0.0f ||
-                normalize_param_.mean_b > 0.0f || normalize_param_.mean_a > 0.0f) {
-              // subtract mean per channel
-              RGBA[0] -= normalize_param_.mean_r;
-              if (n_channels >= 3) {
-                RGBA[1] -= normalize_param_.mean_g;
-                RGBA[2] -= normalize_param_.mean_b;
-              }
-              if (n_channels == 4) {
-                RGBA[3] -= normalize_param_.mean_a;
-              }
-              for (int k = 0; k < n_channels; ++k) {
-                RGBA[k] = RGBA[k] * contrast_scaled + illumination_scaled;
-              }
-            } else if (!meanfile_ready_ || normalize_param_.mean_img.length() == 0) {
-              // do not subtract anything
-              for (int k = 0; k < n_channels; ++k) {
-                RGBA[k] = RGBA[k] * normalize_param_.scale;
-              }
-            } else {
-              CHECK(meanfile_ready_);
-              for (int k = 0; k < n_channels; ++k) {
-                  RGBA[k] = (RGBA[k] - meanimg_[k][i][j]) * contrast_scaled + illumination_scaled;
+            for (int k = 0; k < n_channels; ++k) {
```

Comment:
Is this the same as before? Looks like normalize_param_.scale was only used for meanfile before?

Reply:
The previous one is wrong, if meanfile_ is ready, the normalize_param_.scale was not multiplied.